src/s3_parse_url/storages/amazon.py

```diff
@@ -26,7 +27,7 @@ def _is_host_given(self):
             return False
         return super()._is_host_given()
 
-    def _parse_bucket_name(self) -> str:
+    def _parse_bucket_name(self) -> Optional[str]:
```

Comment:
Actually, I cannot even imagine the situation, when BUCKET is not presented inside  S3 URLs. So, I'm pretty sure, everything related to a bucket should be mandatory rather than optional